RandBLAS/trig_skops.hh

```diff
@@ -316,125 +298,157 @@ namespace RandBLAS {
 
     template <typename T>
     void fht_dispatch(
-        bool left,
+        bool left, // Pre-multiplying?
         blas::Layout layout,
-        T* buff,
-        int64_t log_n,
         int64_t num_rows,
-        int64_t num_cols
+        int64_t num_cols,
+        int64_t log_n,
+        T* A
         )
     {
         if(left && layout == blas::Layout::ColMajor)
-            fht_left_col_major(buff, log_n, num_rows, num_cols);
+            fht_left_col_major(A, log_n, num_rows, num_cols);
         else if(left && layout == blas::Layout::RowMajor)
-            fht_left_row_major(buff, log_n, num_rows, num_cols);
+            fht_left_row_major(A, log_n, num_rows, num_cols);
         else if(!left && layout == blas::Layout::ColMajor)
-            fht_right_col_major(buff, log_n, num_rows, num_cols);
+            fht_right_col_major(A, log_n, num_rows, num_cols);
         else
-            fht_right_row_major(buff, log_n, num_rows, num_cols);
+            fht_right_row_major(A, log_n, num_rows, num_cols);
     }
 }
 
 
 namespace RandBLAS::trig {
+template <SignedInteger sint_t = int64_t>
+struct HadamardMixingOp{
+    sint_t* diag_scale;
+    sint_t* selected_idxs;
```

Comment:
Thinking about it more, I want selected_idxs to always be ``int64_t*``. The reason that sparse matrices use a template parameter ``sint_t`` is because in sparse matrix multiplication moving around indices can actually contribute to the bottleneck runtime. In our context the ``selected_idxs`` is used to specify how we move much larger chunks of data (entire rows in the case of mixing from the left) and so using int64_t doesn't risk making things significantly more expensive.

I'm not sure what to do about the type of ``diag_scale``, to be honest. I won't let this hold back merging the PR but I'd appreciate it if you put thought into this.